src/dune_engine/scheduler.ml

```diff
@@ -1015,9 +1015,11 @@ end = struct
         match t.status with
         | Not_building -> []
         | Building cancellation ->
-          t.handler t.config Build_interrupted;
-          t.status <- Not_building;
-          Fiber.Cancel.fire' cancellation
+          if Fiber.Cancel.fired cancellation
+          then []
+          else (
+            t.handler t.config Build_interrupted;
```

Comment:
Before, we'd call this callback unconditionally. Is there a change of behavior here?

Reply:
Before, it was impossible to hit this case twice in the same build because of the `t.status <- Not_building` line above. But without that line, we can hit this case multiple times, and we only want to sent the `Build_interrupted` notification on the first interruption (instead of sending it on every subsequent interruption).